src/docfx/build/xref/XrefMap.cs

```diff
@@ -14,24 +14,48 @@ namespace Microsoft.Docs.Build
     internal class XrefMap
     {
         // TODO: key could be uid+moniker+locale
-        private readonly IReadOnlyDictionary<string, Lazy<(List<Error>, XrefSpec)>> _internalXrefMap;
+        private readonly IReadOnlyDictionary<string, List<Lazy<(List<Error>, XrefSpec)>>> _internalXrefMap;
         private readonly IReadOnlyDictionary<string, XrefSpec> _externalXrefMap;
         private readonly Context _context;
 
         public IEnumerable<XrefSpec> InternalReferences
-            => _internalXrefMap.Values.Select(v => LoadXrefSpec(v, _context));
+            => _internalXrefMap.Values.SelectMany(v => v.Select(item => LoadXrefSpec(item, _context)));
 
-        public XrefSpec Resolve(string uid)
+        public XrefSpec Resolve(string uid, string moniker = null)
         {
-            if (_internalXrefMap.TryGetValue(uid, out var internalSpec))
+            if (_internalXrefMap.TryGetValue(uid, out var internalSpecs))
             {
-                return LoadXrefSpec(internalSpec, _context);
+                if (!string.IsNullOrEmpty(moniker))
+                {
+                    foreach (var internalSpec in internalSpecs)
+                    {
+                        var spec = LoadXrefSpec(internalSpec, _context);
+                        if (spec.Monikers.Contains(moniker))
+                        {
+                            return spec;
+                        }
+                    }
```

Comment:
should we report a warning if the this uid doesn't contains this moniker?

Reply:
sure